engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/encode/OutputValidator.scala

```diff
@@ -0,0 +1,32 @@
+package pl.touk.nussknacker.engine.avro.encode
+
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+import org.apache.avro.Schema
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.{CustomNodeError, NodeId}
+import pl.touk.nussknacker.engine.api.typed.{CanBeSubclassDeterminer, SingleSubclassRestriction, typing}
+import pl.touk.nussknacker.engine.api.typed.typing.{TypedObjectTypingResult, TypingResult}
+import pl.touk.nussknacker.engine.avro.KafkaAvroBaseTransformer.SinkValueParamName
+import pl.touk.nussknacker.engine.avro.typed.AvroSchemaTypeDefinitionExtractor
+
+object OutputValidator {
+
+  def validateOutput(value: TypingResult, schema: Schema, validationMode: ValidationMode)(implicit nodeId: NodeId): Validated[CustomNodeError, Unit] = {
+    val possibleTypes = AvroSchemaTypeDefinitionExtractor.ExtendedPossibleTypes
+    //FIXME: this still does not handle optional fields validation properly for acceptUnfilledOptional == true?
+    val returnType = new AvroSchemaTypeDefinitionExtractor(skippNullableFields = validationMode.acceptUnfilledOptional).typeDefinition(schema, possibleTypes)
+    val restriction = new SingleSubclassRestriction {
+      override def canBeSubclassOf(givenClass: typing.SingleTypingResult, superclassCandidate: typing.SingleTypingResult): Boolean = (givenClass, superclassCandidate) match {
+        case (TypedObjectTypingResult(objFields, _), TypedObjectTypingResult(superFields, _)) if !validationMode.acceptRedundant =>
+          objFields.keys.forall(superFields.keys.toSet.contains)
```

Comment:
Nice idea, I like this approach - no new mode introduced :+1:

Reply:
After thinking a while:
1. Shouldn't bidirectional check: `value.canBeSubclassOf(returnType) && returnType.canBeSubclassOf(value)` will be enough?
2. Maybe better approch would be just extends some method `checkObjectFields` in `CanBeSubclassDeterminer` instead of creating new abstraction. Current solution will be complicated in the future when we will want to change `&& restriction....` to `|| restriction` or sth similar

Reply:
1. It also came to my mind, but I think cases like: String vs CharSequence, Int vs Long etc. will cause problems 
2. good idea, I'll try it.

Reply:
done